src/chunking.ts

```diff
@@ -0,0 +1,180 @@
+/* eslint-disable no-bitwise, no-plusplus, no-mixed-operators */
+
+import type { Json } from '@metamask/utils';
+import { type TransportChunkFrame } from './types';
+
+const INT32_MAX = 0x7fffffff;
+
+let _randomCounter = 0 | 0;
+
+/**
+ * Creates a random ID for chunked messages. Always <= Int32 max value.
+ *
+ * @returns a random number that can be used as an ID
+ */
+function getNextId(): number {
+  // increment the counter and wrap around if it exceeds Int32 max value
+  _randomCounter = (_randomCounter + 1) & INT32_MAX;
+  return _randomCounter;
+}
+
+// half a "frame" (at 60 fps) per chunk, in milliseconds
+const FRAME_BUDGET_MS = 1000 / 60 / 2;
```

Comment:
Are we likely to hit this during standard usage?

I understood the majority of these messages were the state patch updates from the background RPC to client, so mostly just top level properties from the controllers which I imagine are almost never close to 64MB.

Are we saying this chunking / decoding will likely have a noticeable impact on React responsiveness, or is this a rare fallback?